From a61a1abd887a0d7bad67d5aa17a138deb68e7a44 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 082/296] Catch corruptexception during Transaction operation

If corruption occurs in a transaction running,
DB file should be deleted.
---
 core/java/android/database/sqlite/SQLiteDatabase.java | 9 +++++++++
 1 file changed, 9 insertions(+)
 mode change 100644 => 100755 core/java/android/database/sqlite/SQLiteDatabase.java

diff --git a/core/java/android/database/sqlite/SQLiteDatabase.java b/core/java/android/database/sqlite/SQLiteDatabase.java
old mode 100644
new mode 100755
index df0e262b712..5046f29279f
--- a/core/java/android/database/sqlite/SQLiteDatabase.java
+++ b/core/java/android/database/sqlite/SQLiteDatabase.java
@@ -536,6 +536,9 @@ public final class SQLiteDatabase extends SQLiteClosable {
                             SQLiteSession.TRANSACTION_MODE_IMMEDIATE,
                     transactionListener,
                     getThreadDefaultConnectionFlags(false /*readOnly*/), null);
+        } catch (SQLiteDatabaseCorruptException ex) {
+            onCorruption();
+            throw ex;
         } finally {
             releaseReference();
         }
@@ -549,6 +552,9 @@ public final class SQLiteDatabase extends SQLiteClosable {
         acquireReference();
         try {
             getThreadSession().endTransaction(null);
+        } catch (SQLiteDatabaseCorruptException ex) {
+            onCorruption();
+            throw ex;
         } finally {
             releaseReference();
         }
@@ -666,6 +672,9 @@ public final class SQLiteDatabase extends SQLiteClosable {
         acquireReference();
         try {
             return getThreadSession().yieldTransaction(sleepAfterYieldDelay, throwIfUnsafe, null);
+        } catch (SQLiteDatabaseCorruptException ex) {
+            onCorruption();
+            throw ex;
         } finally {
             releaseReference();
         }
-- 
2.11.0

